test/unit/spec/DeviceFingerprint_spec.js

```diff
@@ -0,0 +1,73 @@
+/*global JSON */
+define([
+  'jquery',
+  'vendor/lib/q',
+  'helpers/util/Expect',
+  'sandbox',
+  'util/DeviceFingerprint'
+],
+function ($, Q, Expect, $sandbox, DeviceFingerprint) {
+
+  Expect.describe('DeviceFingerprint', function () {
+
+    function mockIFrameMessages(success, errorMessage) {
+      var message = success ? {
+        type: 'FingerprintAvailable',
+        fingerprint: 'thisIsTheFingerprint'
+      } : errorMessage;
+      window.postMessage(JSON.stringify(message), '*');
+    }
+
+    beforeEach(function () {
+      $sandbox.append('<div class="auth-content"></div>');
+    });
+
+    it('iframe is created with the right src and it is hidden', function () {
+      spyOn(window, 'addEventListener');
+      DeviceFingerprint.generateDeviceFingerprint('baseUrl');
+      var $iFrame = $('#okta_fingerprint_iframe');
+      expect($iFrame).toExist();
+      expect($iFrame.attr('src')).toBe('baseUrl/auth/services/devicefingerprint');
```

Comment:
I'm assuming this probably will show an error in the console - we had problems with excessive errors etc in the past and ideally we wouldn't introduce more (especially given that they tend to hide real errors, i.e. where we're not handling promises correctly etc).

I think there are examples elsewhere in the test code where we handle this kind of a thing (i.e. overriding network requests) - probably can check the Oauth2 util code